monai/data/wsi_reader.py

```diff
@@ -0,0 +1,288 @@
+            level: the level number, or list of level numbers (default=0)
+            dtype: the data type of output image
+            mode: the output image mode 'RGB' or 'RGBA'
+
+        Returns:
+            a tuples, where the first element is an image [CxHxW], and second element is a dictionary of metadata
+        """
+        # Verify magnification level
+        if level is None:
+            level = self.level
+        max_level = self.get_level_count(wsi) - 1
+        if level > max_level:
+            raise ValueError(f"The maximum level of this image is {max_level} while level={level} is requested)!")
+
+        # Verify location
+        if location is None:
+            location = (0, 0)
+        wsi_size = self.get_size(wsi, level)
+        if location[0] > wsi_size[0] or location[1] > wsi_size[1]:
+            raise ValueError(f"Location is outside of the image: location={location}, image size={wsi_size}")
+
+        # Verify size
+        if size is None:
+            if location != (0, 0):
+                raise ValueError("Patch size should be defined to exctract patches.")
+            size = self.get_size(wsi, level)
+        else:
+            if size[0] <= 0 or size[1] <= 0:
+                raise ValueError(f"Patch size should be greater than zero, provided: patch size = {size}")
+
+        # Extract a patch or the entire image
+        patch = self.get_patch(wsi, location=location, size=size, level=level, dtype=dtype, mode=mode)
+
+        # Verify patch image
+        patch = self.verify_output(patch, mode)
+
+        # Set patch-related metadata
+        metadata = self.get_metadata(wsi=wsi, patch=patch, location=location, size=size, level=level)
+
+        return patch, metadata
+
+    def verify_suffix(self, filename: Union[Sequence[PathLike], PathLike]) -> bool:
+        """
+        Verify whether the specified file or files format is supported by WSI reader.
+
+        Args:
+            filename: file name or a list of file names to read.
+                if a list of files, verify all the suffixes.
+        """
+        return is_supported_format(filename, self.supported_formats)
+
+    def verify_output(self, patch: np.ndarray, mode: str):
+        """
+        Verify output image patch
+        """
+        # check if the image has three dimensions (2D + color)
+        if patch.ndim != 3:
+            raise ValueError(
+                f"The image dimension should be 3 but has {patch.ndim}. "
+                "`WSIReader` is designed to work only with 2D images with color channel."
+            )
+
+        return patch
+
+
+class WSIReader(BaseWSIReader):
+    """
+    WSIReader that supports different implemented backends
+
+    The support for any backend can be achieved by
+
+    .. code-block:: python
+
+        if self.backend == "any_backend":
+            self.backend_lib = AnyBackendWSIReader(level=level, **kwargs)
+
+    """
+
+    def __init__(self, backend="cucim", level: int = 0, **kwargs):
```

Comment:
Although this class is kind of `wrapper`, I still suggest to add doc-strings for every function, because user will only check the doc-strings of this class when using it.

Thanks.

Reply:
Could you please help also add the doc-strings of `args` for this init function? `backend` and `level`, `kwargs`.

Thanks.